core/domain/user_services_test.py

```diff
@@ -1150,3 +1153,179 @@ def test_last_exp_edit_time_gets_updated(self):
+
+        logging_swap = self.swap(logging, 'error', _mock_logging_function)
+        assert_raises_regexp_exception = self.assertRaisesRegexp(
+            Exception, 'User not found.')
+
+        with logging_swap, assert_raises_regexp_exception:
+            user_services.get_human_readable_user_ids(['invalid_user_id'])
+
+        self.assertEqual(
+            observed_log_messages,
+            [
+                'User id invalid_user_id not known in list of user_ids '
+                '[\'invalid_user_id\']'
+            ])
+
+    def test_get_human_readable_user_ids(self):
+        user_ids = user_services.get_human_readable_user_ids(
+            [self.owner_id, feconf.SYSTEM_COMMITTER_ID, self.guest_id])
+        expected_user_ids = [
+            'owner', 'admin',
+            '[Awaiting user registration: new.u..@example.com]']
+
+        self.assertEqual(user_ids, expected_user_ids)
+
+
+class UserContributionsTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(UserContributionsTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.user_contributions = user_services.get_user_contributions(
+            self.owner_id)
+        self.user_contributions.validate()
+
+    def test_validate_non_str_user_id(self):
+        self.user_contributions.user_id = 0
+        with self.assertRaisesRegexp(
+            Exception, 'Expected user_id to be a string'):
+            self.user_contributions.validate()
+
+    def test_validate_user_id(self):
+        self.user_contributions.user_id = ''
+        with self.assertRaisesRegexp(Exception, 'No user id specified.'):
+            self.user_contributions.validate()
+
+    def test_validate_non_list_created_exploration_ids(self):
+        self.user_contributions.created_exploration_ids = 0
+        with self.assertRaisesRegexp(
+            Exception, 'Expected created_exploration_ids to be a list'):
+            self.user_contributions.validate()
+
+    def test_validate_created_exploration_ids(self):
+        self.user_contributions.created_exploration_ids = [0]
+        with self.assertRaisesRegexp(
+            Exception, 'Expected exploration_id in created_exploration_ids '
+            'to be a string'):
+            self.user_contributions.validate()
+
+    def test_validate_non_list_edited_exploration_ids(self):
+        self.user_contributions.edited_exploration_ids = 0
+        with self.assertRaisesRegexp(
+            Exception, 'Expected edited_exploration_ids to be a list'):
+            self.user_contributions.validate()
+
+    def test_validate_edited_exploration_ids(self):
+        self.user_contributions.edited_exploration_ids = [0]
+        with self.assertRaisesRegexp(
+            Exception, 'Expected exploration_id in edited_exploration_ids '
+            'to be a string'):
+            self.user_contributions.validate()
```

Comment:
Add "is a valid value" tests.

Reply:
It is already validated in the setup function [here](https://github.com/oppia/oppia/pull/6980/files/110a36aade1d595ded3aa69dfb2df204e0525fa2#diff-3c9fdf28b2537db9d7ab417f72581883R1268)